dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedFileWriter.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.client.file.CacheContext;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.exception.runtime.InternalRuntimeException;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.worker.block.io.BlockWriter;
+
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A paged implementation of BlockWriter interface.
+ */
+public class PagedFileWriter extends BlockWriter {
+  private static final Logger LOG = LoggerFactory.getLogger(PagedFileWriter.class);
+  private final CacheContext mTempCacheContext;
+
+  private final CacheManager mCacheManager;
+  private final String mFileId;
+  private final long mPageSize;
+  private long mPosition;
+
+  PagedFileWriter(CacheManager cacheManager, String fileId, long pageSize) {
+    mTempCacheContext = CacheContext.defaults().setTemporary(true);
+    mCacheManager = cacheManager;
+    mFileId = fileId;
+    mPageSize = pageSize;
+  }
+
+  @Override
+  public void commitFile() {
+    mCacheManager.commitFile(mFileId);
+  }
+
+  @Override
+  public long append(ByteBuffer inputBuf) {
+    long bytesWritten = 0;
+    while (inputBuf.hasRemaining()) {
+      PageId pageId = getPageId(bytesWritten);
+      int currentPageOffset = getCurrentPageOffset(bytesWritten);
+      int bytesLeftInPage = getBytesLeftInPage(currentPageOffset, inputBuf.remaining());
+      byte[] page = new byte[bytesLeftInPage];
+      inputBuf.get(page);
+      if (!mCacheManager.append(pageId, currentPageOffset, page, mTempCacheContext)) {
+        throw new InternalRuntimeException("Append failed for block " + mFileId);
+      }
+      bytesWritten += bytesLeftInPage;
+    }
+    mPosition += bytesWritten;
+    return bytesWritten;
+  }
+
+  @Override
+  public long append(ByteBuf buf) throws IOException {
+    long bytesWritten = 0;
+    while (buf.readableBytes() > 0) {
+      PageId pageId = getPageId(bytesWritten);
```

Comment:
s/bytesWritten/mPosition?

Should we use the position to determine the current page?